packages/google_adsense/README.md

```diff
@@ -0,0 +1,60 @@
+# google_adsense
+[Google AdSense](https://adsense.google.com/intl/en_us/start/) plugin for Flutter Web
+
+Generally integration with AdSense requires:
+
+1. Adding an [AdSense code](https://support.google.com/adsense/answer/9274634?hl=en&ref_topic=28893&sjid=9002959242386336076-EU) usually in between the `<head></head>` tag to connect your site with AdSense account. Allows using **Auto ads**
+2.  Adding [ad unit code](https://support.google.com/adsense/answer/9274025?sjid=9002959242386336076-EU) in between the `<body><body/>` tags of your pages where you want ad to appear. Ad unit code snippet is unique per **ad unit** that you need to create in your AdSense account first. You might have several ad units added to your website
+
+Since Flutter apps are not rendered as traditional HTML pages and there is no easy way to access final DOM tree from Dart, this plugin provides an `AdUnitWidget` that you can configure and place in the desired location in your app UI
+
+
+## Installation
+run `flutter pub add google_adsense`
+
+## Usage
+#### Initialize AdSense
+Before displaying ads, initialize the AdSense with your ad client ID.
+<?code-excerpt "example/lib/main.dart (init)"?>
+```dart
+import 'package:google_adsense/google_adsense.dart';
+
+void main() {
+  Adsense().initialize('your_ad_client_id');
+  runApp(const MyApp());
+}
+
+```
+#### Display AdUnitWidget
+<?code-excerpt "example/lib/main.dart (adUnit)"?>
+```dart
+Adsense().adUnit(
+    adSlot: 'your_ad_slot_id',
+    isAdTest: true,
+    adUnitParams: <String, dynamic>{
+      AdUnitParams.AD_FORMAT: 'auto',
+      AdUnitParams.FULL_WIDTH_RESPONSIVE: true,
+    },
```

Comment:
I created `AdUnitParams` class just to provide String keys for **some** of the common parameters that developers might want to pass to adUnitParams argument. There might be more of those

But, I do think there is no need for dynamic, it can be strongly typed String. Under the hood these will simply become data attributes on the html element. E.g.  `AdUnitParams.AD_FORMAT: 'auto'` becomes `data-ad-format=auto`